src/domain/prompt/IDirectorySelectionPromptConfig.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Flow Interoperability Project (flowinterop.org)
+ * Flow Runner
+ * Copyright (c) 2019, 2020 Viamo Inc.
+ * Authored by: Brett Zabos (brett.zabos@viamo.io)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ **/
+
+import {IPromptConfig, IChoice} from '../..'
+
+export interface IDirectorySelectionPromptConfig extends IPromptConfig<string | null> {
```

Comment:
I think the generic here is wrong. The way it works on ClipboardWeb, is the type for the value is `List<DirectorySelection>` where DirectorySelection is the following object that can be represented as a JSONObject:
```
data class DirectorySelection(val name: String, val value: String)
```
If you want to see the code for this, it's in https://bitbucket.org/voto/clipboardcore/src/110dbb227cfb0c5e3c6e1ec1e4ea60fdc79acf1f/src/commonMain/kotlin/viamo/model/block/ExecutionState.kt#ExecutionState.kt-183

Reply:
@AlteaDown why is a `List<DirectorySelection>` used, and not just a row index number?

Reply:
Could we do something like:

```
export interface IDirectorySelectionPromptConfig extends IPromptConfig<number> {
  rows: ChoiceRow[],
}
interface ChoiceRow {
  primaryText: string,
  secondaryTexts: string[],
}
```

Reply:
I actually can't remember the reasoning. However, this is the form that the DB has BlockInteractions of right now, so they should be the same.

Reply:
Maybe to persist the right selection if the rows get shuffled = if the directory is updated (but still has the same columns)

Reply:
Likely; sometimes we include additional details so the interaction can be more self-describing and debuggable.

Reply:
There's no requirement for uniqueness, so I think that was it